### PR TITLE
[TE] Self-Serve Create Alert: Prelaunch Fixes

### PR DESCRIPTION
**Pull request by newsummit**
Changed:
- Conditional display of dimension options (based on selected filters)
- Minutely metrics now support dimensional filters
- Change alert property name "anomalyRemovalThreshold"
- Some style changes for success/failure notifications
- Triggering replay with new unified endpoints

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js [hunk 1]
```diff
@@ -256,37 +258,70 @@ export default Ember.Controller.extend({
...
+  /**
+   * Replay Flow Step 2 - Replay cloned function
+   * @method callReplayStart
+   * @param {Number} clonedId - id of the cloned function
+   * @param {Object} startTime - replay start time stamp
+   * @param {Object} endTime - replay end time stamp
+   * @return {Ember.RSVP.Promise}
+   */
+  callReplayStart(clonedId, startTime, endTime) {
+    const url = `/detection-job/${clonedId}/replay?start=${startTime}&end=${endTime}&speedup=true`;
```
Comment (cecilynie) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js]: Hi Steve, here we have a different configuration: hourly and daily metrics, speedup should be "false".

File: thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js [hunk 2]
```diff
@@ -209,25 +211,25 @@ export default Ember.Controller.extend({
...
    */
   triggerGraphFromMetric(metricId) {
+    const id = metricId.id;
     const maxTime = this.get('maxTime');
-    const filters = this.get('filters');
     const currentEnd = moment(maxTime).isValid()
       ? moment(maxTime).valueOf()
       : moment().subtract(1, 'day').endOf('day').valueOf();
-    const granularity = this.get('selectedGranularity') || this.get('granularities.firstObject');
+    const filters = this.get('selectedFilters') || '';
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js]: fyi we can also use `getWithDefault` here. https://www.emberjs.com/api/ember/2.14/classes/Ember.Object/methods/getWithDefault?anchor=getWithDefault

File: thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js [hunk 3]
```diff
@@ -256,37 +258,70 @@ export default Ember.Controller.extend({
...
+    this.callCloneAlert(newFuncId)
+      .then((clonedId) => {
+        this.set('isReplayTriggeredSuccess', true);
+        return this.callReplayStart(clonedId, startTime, endTime);
+      })
+      .catch((error) => {
+        this.setErrorState(error, 'Failed to clone new alert function');
+      })
+      .then((jobId) => {
+        this.set('isReplayStatusPending', false);
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js]: this `.then` would only be called if an error was caught on line 315. Is this the intended behavior?

File: thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js (hunk 1 above)
Comment (newsummit) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js]: got it. thanks.

File: thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js (hunk 3 above)
Comment (newsummit) [inline thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js]: No, the first catch is for "callCloneAlert" and the second one is for "callReplayStart". It is properly entering the ".then" after "callCloneAlert" resolves. I've tested both error handlers and they work as expected.

**Pull request merged**